metaflow/plugins/aws/step_functions/dynamo_db_client.py

```diff
@@ -65,6 +66,16 @@ def get_parent_task_ids_for_foreach_join(self,
         return response['Item']['parent_task_ids_for_foreach_join']['SS']
 
     def _get_instance_region(self):
-        return os.popen(
-            'curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone/'
-            ).read()[:-1]
\ No newline at end of file
+        try:
+            r = requests.post(
+                url = "http://169.254.169.254/latest/meta-data/placement/availability-zone/"
+            )
+            if r.status_code != 200:
+                raise RuntimeError("Failed to query AWS region from " +
+                        "http://169.254.169.254/latest/meta-data/placement/availability-zone/\n" +
```

Comment:
`("Failed to query instance metadata. Url [%s], Error code [%s]" % (url, r.status_code)` would be cleaner.

Reply:
@jasonge27 Let's address this comment and then you should be good to merge.